### PR TITLE
test: Print --help in check-verify and check-* tests properly

### PR DESCRIPTION
**Pull request by stefwalter**
Parse the args in a more coherent way.

### CODE REVIEW

**Comment by dperpeet** [comment]
needs a rebase

**Comment by mvollmer** [comment]
The abort happens because we use the new testlib.py with the old testinfra.py.

**Comment by mvollmer** [comment]
Testing is stuck on this because PRs with status 'error' are retried with their normal priority.

File: test/testlib.py [hunk 1]
```diff
@@ -970,44 +968,32 @@ def test_main(argv=None, suite=None, attachments=None):
...
-    parser.add_argument('-s', dest='sit', action='store_true')
-    parser.add_argument('tests', nargs='*')
-
-    parser.set_defaults(verbosity=1)
-    args = parser.parse_args(argv)
-
-    if args.sit and args.jobs > 1:
+    standalone = opts is None
+    if standalone:
+        parser = testinfra.arg_parser()
```
Comment (mvollmer) [inline test/testlib.py]: Now `parser` is only defined when running standalone.  `check-verify foo` will thus crash when trying to raise "tests may not be specified when ...".

**Comment by stefwalter** [comment]
> Now parser is only defined when running standalone. check-verify foo will thus crash when trying to raise "tests may not be specified when ...".

Not unconditionally creates ```parser```

**Pull request closed without merging**